src/py/pyodide/console.py

```diff
@@ -479,19 +479,66 @@ async def runcode(self, source: str, code: CodeRunner) -> ConsoleFuture:
         return await super().runcode(source, code)
 
 
-def repr_shorten(
-    value: Any, limit: int = 1000, split: int | None = None, separator: str = "..."
+def shorten(
+    text: str, limit: int = 1000, split: int | None = None, separator: str = "..."
 ) -> str:
-    """Compute the string representation of ``value`` and shorten it
-    if necessary.
+    """Shorten ``text`` it if necessary.
 
     If it is longer than ``limit`` then return the firsts ``split``
     characters and the last ``split`` characters separated by '...'.
     Default value for ``split`` is `limit // 2`.
+    If ``split`` is longer than the length of ``len(value) // 2``, ``split`` will be `len(value) // 2`.
+    limit must be greater than or equal to 2. If it is less than 2, an error occurs.
```

Comment:
Some edits:
```suggestion
    """Shorten ``text`` if it is longer than ``limit``.

	If ``len(text) <= limit`` then return ``text`` unchanged.
    If ``text`` is longer than ``limit`` then return the firsts ``split``
    characters and the last ``split`` characters separated by ``separator``.
    The default value for ``split`` is `limit // 2`.
    Values of ``split`` larger than ``len(value) // 2`` will have the same effect as 
    when ``split`` is `len(value) // 2`.
    A value error is raised if ``limit`` is less than 2.
```
Perhaps we should add `Parameters` and `Returns` sections like the other functions in this file?